### PR TITLE
Fix the FORM package

### PR DESCRIPTION
**Pull request by tueda**
- The first commit aims to fetch the correct tarballs.
- The second one fixes the dependency.

---

The first one may need some explanation. Currently, `form` executables installed via Spack do not know revision information correctly. Indeed, the build dates are printed:
```
# form@4.2.1
$ form -v
FORM 4.2 (Oct  2 2021) 64-bits
  0.00 sec out of 0.00 sec
```
```
# form@4.1-20131025
$ form -v
FORM 4.1 (Oct  2 2021) 64-bits
  0.00 sec out of 0.00 sec
```
This is because the fetched tarballs are not intended to be used for build and do not contain detailed revision information. You see some warnings about the revision information on running `spack install -v form@4.2.1`:
```
========================================================================
Failed to determine the revision of the source code.

The reason may be
  - this is neither a source distribution (containing the configure
    script) nor a cloned Git repository,
  - this is a shallow clone and no version tags are reachable,
  - some required utilities (e.g., git) are missing.

Source distributions and some binaries can be found in:

  http://www.nikhef.nl/~form/maindir/binaries/binaries.html
  https://github.com/vermaseren/form/releases

The latest source code can be cloned by:

  git clone https://github.com/vermaseren/form.git

You can continue the build, but binaries will not contain the revision
information.
========================================================================
```
After applying the first commit, `form` prints the revision information correctly:
```
# form@4.2.1
$ form -v
FORM 4.2.1 (Nov 21 2018, v4.2.1) 64-bits
  0.00 sec out of 0.00 sec
```
```
# form@4.1-20131025
$ form -v
FORM 4.1 (Oct 25 2013) 64-bits
  0.00 sec out of 0.00 sec
```

### CODE REVIEW

File: var/spack/repos/builtin/packages/form/package.py
Comment (iarspider) [inline var/spack/repos/builtin/packages/form/package.py]:
```suggestion
    version('4.1-20131025', sha256='fb3470937d66ed5cb1af896b15058836d2c805d767adac1b9073ed2df731cbe9',
            url='https://github.com/vermaseren/form/releases/download/v4.1-20131025/form-4.1.tar.gz')
```

and remove `url_for_version` function

File: var/spack/repos/builtin/packages/form/package.py
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/form/package.py]:
```suggestion
```
Since you switched to release tarballs which contain a `configure` script, `AutotoolsPackage` no longer needs to generate it using `autoreconf`, so the autotools dependencies are obsolete (including m4), tested it. You can just accept the suggestion using the button below. With that, added, you can request a review from me.

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
Please apply my suggestion to remove the 4 autoreconf dependencies, tested it and I was co-authoring the last updated of the related code in the AutotoolsPackage: t is ok to remove these for the release tarballs (not used for them).

File: var/spack/repos/builtin/packages/form/package.py
Comment (tueda) [reply var/spack/repos/builtin/packages/form/package.py]: Thank you for the review. I'm fine with this.

**Comment by tueda** [comment]
Another point: can we switch `gmp` and `zlib` variants to `default=True`? (I'm not sure whether there is some Spack community policy like default variants should be minimal.) As written in the [FORM Wiki](https://github.com/vermaseren/form/wiki/Installation#building-from-the-source-distribution), they are recommended to use:
> We recommend building FORM with two external libraries, GMP and zlib.

**Review by iarspider — CHANGES_REQUESTED** [review]

**Comment by iarspider** [comment]
Sorry, I thought my review was visible. One minor change, otherwise LGTM

File: var/spack/repos/builtin/packages/form/package.py
Comment (tueda) [reply var/spack/repos/builtin/packages/form/package.py]: Thank you for your review. I will make a change following your suggestion.

Comment (iarspider) [reply var/spack/repos/builtin/packages/form/package.py]: You don't need `url_for_version` function if you pass `url` parameter

**Comment by bernhardkaindl** [comment]
> Another point: can we switch `gmp` and `zlib` variants to `default=True`? (I'm not sure whether there is some Spack community policy like default variants should be minimal.) As written in the [FORM Wiki](https://github.com/vermaseren/form/wiki/Installation#building-from-the-source-distribution), they are recommended to use:
> 
> > We recommend building FORM with two external libraries, GMP and zlib.

Good reference! :+1:  By all means, and because of gmp and zlib are not heavvy dependencies, it would be ok for me.

But I'm not the maintainer, I don't use this package, so I'm referring this to @iarspider for his take!

@tueda, from the feel of it, you seem to care about the package, so if ok for @iarspider, would be be content to add yourself as a 2nd maintainer after iarspider? You'd only be notified by spackbot like iarspider was when somebody submits a PR, there is no mandatory response needed on your part, just people would know that you can be asked if there is a question or need of a review.

@iarspider, what do you think, is it worth it enable those by default (and maybe if ok for him, have him as 2nd maintainer) ?

File: var/spack/repos/builtin/packages/form/package.py
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/form/package.py]: AFAICS, this is pushed now as commmit 67165dd by Takahiro

**Review by bernhardkaindl — DISMISSED** [review]
If you like to be finished, you can just push enabing gmp and zlib by default and wait for iarspider's opinion/review on it.

**Review by iarspider — DISMISSED** [review]

File: var/spack/repos/builtin/packages/form/package.py
Comment (iarspider) [reply var/spack/repos/builtin/packages/form/package.py]: Yup. Still getting used to reviewing, sorry!

**Review by bernhardkaindl — DISMISSED** [review]
Since I saw that iarspider approved this commit, let's merge it! :100:

**Comment by tueda** [comment]
Rebased with the removal of `url_for_version`. (I hope it works with your co-authorship.)
I also added the 3rd commit to enable `gmp` and `zlib` by default.

If you guys are OK, I am fine to become the 2nd maintainer.

**Comment by iarspider** [comment]
I'm fine with both changes.

File: var/spack/repos/builtin/packages/form/package.py
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/form/package.py]:
```suggestion
    maintainers = ['iarspider', 'tueda']
```
@iarspider I guess this also meant your are fine with this too? (you can commit my suggestion with the button above)

Comment (tueda) [reply var/spack/repos/builtin/packages/form/package.py]: Sure.

Comment (iarspider) [reply var/spack/repos/builtin/packages/form/package.py]: Yes - both adding tueda as a maintainer and changing gmp and zlib defaults

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/form/package.py]: Done, that's all from me as well! :+1: Thanks!

**Review by bernhardkaindl — APPROVED** [review]

File: var/spack/repos/builtin/packages/form/package.py
Comment (tueda) [reply var/spack/repos/builtin/packages/form/package.py]: Oh, sorry, I pushed the button without reading your text carefully...

**Review by iarspider — APPROVED** [review]

**Comment by bernhardkaindl** [comment]
Done! :1st_place_medal: :-)

In case the check `ci/gitlab-ci`, fails that's not fatal! (it has problems not related to this package and is being worked on).
We can merge without `ci/gitlab-ci` at the moment.

In case you wonder about the commit message:
For the upcoming auto-squash, I updated it try to follow the usual commit message style for spack with all commit comments below that squashed to just the Co-autored lines.

**Thanks for the contribution, to be fine to be listed as 2nd maintainer and thanks for the review with suggestion iarspider!**

**Comment by bernhardkaindl** [comment]
Looks good, I've built the final merged commit from develop, *I think that's it for this one!* :+1:

**Pull request merged**
